cardano-config/src/Cardano/TracingOrphanInstances/Common.hs

```diff
@@ -70,3 +76,196 @@ deriving newtype instance FromJSON EpochNo
 instance ToObject Void where
   toObject _verb x = case x of {}
 
+-- Common failures
+
+instance ToObject (PredicateFailure (DELEG c)) where
+  toObject _verb (StakeKeyAlreadyRegisteredDELEG alreadyRegistered) =
+    mkObject [ "kind" .= String "StakeKeyAlreadyRegisteredDELEG"
+             , "Credential" .= String (textShow alreadyRegistered)
+             , "error" .= String "Staking credential already registered"
+             ]
+  toObject _verb (StakeKeyNotRegisteredDELEG notRegistered) =
+    mkObject [ "kind" .= String "StakeKeyNotRegisteredDELEG"
+             , "Credential" .= String (textShow notRegistered)
+             , "error" .= String "Staking credential not registered"
+             ]
+  toObject _verb (StakeKeyNonZeroAccountBalanceDELEG remBalance) =
+    mkObject [ "kind" .= String "StakeKeyNonZeroAccountBalanceDELEG"
+             , "Stake key remaining balance, if it exists" .= String (textShow remBalance)
+             ]
+  toObject _verb (StakeDelegationImpossibleDELEG unregistered) =
+    mkObject [ "kind" .= String "StakeDelegationImpossibleDELEG"
+             , "Unregistered credential" .= String (textShow unregistered)
+             , "error" .= String "Cannot delegate this stake credential because it is not registered"
+             ]
+  toObject _verb WrongCertificateTypeDELEG =
+    mkObject [ "kind" .= String "WrongCertificateTypeDELEG" ]
+  toObject _verb (GenesisKeyNotInpMappingDELEG (KeyHash genesisKeyHash)) =
+    mkObject [ "kind" .= String "GenesisKeyNotInpMappingDELEG"
+             , "Unknown genesis key hash" .= String (textShow genesisKeyHash)
+             , "error" .= String "This genesis key is not in the delegation mapping"
+             ]
+  toObject _verb (DuplicateGenesisDelegateDELEG (KeyHash genesisKeyHash)) =
+    mkObject [ "kind" .= String "DuplicateGenesisDelegateDELEG"
+             , "Duplicate genesis key hash" .= String (textShow genesisKeyHash)
+             , "error" .= String "This genesis key has already been delegated to"
+             ]
+  toObject _verb (InsufficientForInstantaneousRewardsDELEG neededMirAmount reserves) =
+    mkObject [ "kind" .= String "InsufficientForInstantaneousRewardsDELEG"
+             , "neededMirAmount" .= neededMirAmount
```

Comment:
For example here we don't need "Mir" in the field name, because it's clear from the context.